### DISCUSSION TITLE
Add sensors to an existing product before manufacturing

### DISCUSSION THREAD

**Discussion by risoul** [arendst/Tasmota#21450] [Support request] [answered]
Hi guys,

I have a project based on the PG01V3 : https://templates.blakadder.com/athom_PG01V3-EU16A.html

I would like to modify hardware and code to add DS18B20 sensors + 2 waterflow sensors YF-S201.

The goal is to produce and distribute the modified product to my clients.

Questions:
- do you think there is enough place in the case to add 3 RJ45 female connectors ?
- is there 3 free GPIO pins that can read sensors ?
- are GPIO easy to access to add connections to RJ45 plugs ?
- what are your though about commercialization of a modified products (licences, patents...) ?
- any tip about manufacturing ?

Thank you

**Comment by sfromis** [comment, marked as answer]
The gpios you have are those on the small green board minus ones already used. You do the counting..... To me, it looks "questionable".

Doing such modification in a safe way may be "less feasible", as gouging holes will lose structural strength, and finding good places for quite bulky plugs (RJ45 is a "huge" beast) is not realistic.

What's even worse is that you should **never** add stuff to energy monitoring plugs, as this alone is a major safety hazard.

IMO, bad plan.

**Reply by barbudor** [reply]
+1

Just don't do it
Electric shock ahead
For tinkering hardware, use appropriate boards with isolated power
On power monitoring plus, all the circuitry is at mains level

**Reply by risoul** [reply]
Great answers, thank you guys. You are right, it was looking as a good idea but it is not...
Will go back to my project with M5stack core witch is easier to customize.
